src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -227,6 +227,11 @@ protected virtual bool TryGetKeyIdentifierHeader(out string keyIdentifier, IInco
             return context.Headers.TryGetValue(Headers.RijndaelKeyIdentifier, out keyIdentifier);
         }
 
+        protected virtual void GenerateIV(RijndaelManaged rijndael)
```

Comment:
The rationale here is to enable derived classes to ask `RijndaelManaged` instances for various parameters like mode, block size etc. to generate a proper IV. This is now only exposed internally to aid testability but could become more relevant if we break encryption out of the core and allow more customization by users.